### PR TITLE
[improve][broker] Get lowest PositionImpl from NavigableSet

### LINKED ISSUE #18279: [cleanup][broker] Direct get lowest PositionImpl from TreeMap
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

some interface already return TreeSet but the signature is Set 
so the caller will use set.stream().findFirst() 
which is unnessary.

### Solution

_No response_

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by WJL3333**
Fixes #18279

### Motivation
change signature from `Set<T>` to `NavigableSet<T>`
which makes the caller to get lowest `PositionImpl` more efficient ( compare with `stream().findfirst()` )

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is already covered by existing tests, such as *(please describe tests)*.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by WJL3333** [comment]
@codelipenghui @Jason918 please take a look, thank you~

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java [hunk 1]
```diff
@@ -285,7 +286,7 @@ public synchronized void readMoreEntries() {
                 }
 
                 havePendingReplayRead = true;
-                minReplayedPosition = messagesToReplayNow.stream().min(PositionImpl::compareTo).orElse(null);
+                minReplayedPosition = messagesToReplayNow.pollFirst();
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java:289]: It should be first(), not pollFirst()?

**Review by codelipenghui — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java [hunk 2]
```diff
@@ -285,7 +286,7 @@ public synchronized void readMoreEntries() {
                 }
 
                 havePendingReplayRead = true;
-                minReplayedPosition = messagesToReplayNow.stream().min(PositionImpl::compareTo).orElse(null);
+                minReplayedPosition = messagesToReplayNow.first();
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java:289]: You need to check the `messagesToReplayNow.isEmpty()`.

Comment (WJL3333) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java:289]: thanks for review, `messagesToReplayNow.isEmpty()` is called in line 282

**Review by coderzc — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by nodece — APPROVED** [review]

**Review by labuladong — APPROVED** [review]

**Pull request merged**